spec/function.dd

```diff
@@ -271,6 +271,9 @@ pure List* make(int a, int b)
     assumptions related to memoization and references do not carry to the thrown
     exception.)
 
+    $(P Destructors will always be called even if they appear to be strongly
```

Comment:
"Destructors will always be called" is not want you want to say I think; instead, it should only say something about pure destructors.
Perhaps: "When a destructor is pure, it does not imply that it need not be called."


Maybe I am approaching this from too low-level viewpoint. When you write "destructors will always be called", do you mean it in the sense of debug-call-tracing-low-levelness, or in the high level sense of "the program will be executed as if destructors are always called" (thus allowing empty destructor calls to be eliminated)?